src/main/java/io/stargate/sgv2/jsonapi/service/embedding/gateway/EmbeddingGatewayClient.java

```diff
@@ -0,0 +1,162 @@
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setIntValue((Integer) value)
+                      .build());
+            else if (value instanceof Float)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setFloatValue((Float) value)
+                      .build());
+            else if (value instanceof Boolean)
+              grpcVectorizeServiceParameter.put(
+                  key,
+                  EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.ParameterValue.newBuilder()
+                      .setBoolValue((Boolean) value)
+                      .build());
+          });
+    }
+    EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest embeddingRequest =
+        EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.newBuilder()
+            .setModelName(modelName)
+            .setDimensions(dimension)
+            .putAllParameters(grpcVectorizeServiceParameter)
+            .setInputType(
+                embeddingRequestType == EmbeddingRequestType.INDEX
+                    ? EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.InputType.INDEX
+                    : EmbeddingGateway.ProviderEmbedRequest.EmbeddingRequest.InputType.SEARCH)
+            .addAllInputs(texts)
+            .build();
+
+    final EmbeddingGateway.ProviderEmbedRequest.ProviderContext.Builder builder =
+        EmbeddingGateway.ProviderEmbedRequest.ProviderContext.newBuilder()
+            .setProviderName(provider)
+            .setTenantId(tenant.orElse("default"));
+    if (apiKeyOverride.isPresent()) {
+      builder.putAuthTokens("API_KEY", apiKeyOverride.orElse(apiKey));
+    }
+
+    EmbeddingGateway.ProviderEmbedRequest.ProviderContext providerContext = builder.build();
+    EmbeddingGateway.ProviderEmbedRequest providerEmbedRequest =
+        EmbeddingGateway.ProviderEmbedRequest.newBuilder()
+            .setEmbeddingRequest(embeddingRequest)
+            .setProviderContext(providerContext)
+            .build();
+
+    final Uni<EmbeddingGateway.EmbeddingResponse> embeddingResponse =
+        embeddingService.embed(providerEmbedRequest);
+    return embeddingResponse
+        .onItem()
+        .transform(
+            resp -> {
+              if (resp.hasError()) {
+                throw new JsonApiException(
+                    ErrorCode.valueOf(resp.getError().getErrorCode()),
+                    resp.getError().getErrorMessage());
+              }
+              if (resp.getEmbeddingsList() == null) {
+                return Collections.emptyList();
+              }
+              return resp.getEmbeddingsList().stream()
+                  .map(
+                      data -> {
+                        Float[] embedding = new Float[data.getEmbeddingList().size()];
```

Comment:
`Float[]` is quite inefficient (wrapper over `float`) -- is there any chance native `float[]` could be used instead? This would avoid construction of throw-away `Float` values if input is as `float`s.